crates/ruma-common/src/events/message/content_serde.rs

```diff
@@ -49,13 +67,31 @@ impl Serialize for MessageContent {
     where
         S: serde::Serializer,
     {
-        let mut st = serializer.serialize_struct("MessageContent", 1)?;
-        if self.len() == 1 && self[0].mimetype == "text/plain" {
-            st.serialize_field("org.matrix.msc1767.text", &self[0].body)?;
+        if self.iter().all(|message| {
+            #[cfg(feature = "unstable-msc3554")]
+            {
+                matches!(&*message.mimetype, "text/plain" | "text/html") && message.lang.is_none()
+            }
+
+            #[cfg(not(feature = "unstable-msc3554"))]
+            {
+                matches!(&*message.mimetype, "text/plain" | "text/html")
+            }
+        }) {
```

Comment:
Can you move this closure out of the `if` expression? Maybe as two separate cfg'd `fn` items, or `let` bindings.